docs/implementations/shell.md

```diff
@@ -0,0 +1,87 @@
+# Shell-imp
+
+The `shell-imp` is a wrapper enabling models external to IEF to be executed as part of an IEF pipeline. For example, you might have a standalone model written in Python. `shell-imp` spawns a subprocess to execute that Python model in a dedicated shell and pipes the results back into IEF's Typescript process.
+
```

Comment:
```suggestion

## Parameters
### Model config
The model should be intialized with a name and `kind: shell`, as follows:

The shell model interface requires a path to the model executable. This path is provided in the model configuration with the name `executable`. The path should be appended by the execution command, for example if the executable is a binary, the path would be prepended with `./` on a Linux system. If the model is intended to be run as Python, you can prepend `python`.

- `executable`: the path to the model executable along with the execution command as it would be entered into a shell.

### inputs
The parameters included in the `inputs` field int he `impl` depend entirely on the model itself. A typical model plugin might expect the following common data to be provided as `inputs`: 
- `timestamp`: A timestamp for the specific input observation
- `duration`: The length of time these specific inputs cover
- `cpu-util`: The percentage CPU utilization for the specific duration.

## Returns

The specific return types depend on the model being invoked. Typically, we would expect some kind of energy or carbon metric as an output, but it is also possible that models target different parts of the pipeline, such as data importers, adaptor models etc. Therefore, we do not specify return data for external models.

```